packages/apollo-link-schema/src/__tests__/schemaLink.ts

```diff
@@ -0,0 +1,115 @@
+import { Observable, ApolloLink, execute } from 'apollo-link';
+import { print, graphql } from 'graphql';
+import { makeExecutableSchema } from 'graphql-tools';
+import gql from 'graphql-tag';
+
+import { SchemaLink } from '../schemaLink';
+
+const sampleQuery = gql`
+  query SampleQuery {
+    stub {
+      id
+    }
+  }
+`;
+
+const sampleMutation = gql`
+  mutation SampleMutation {
+    stub(param: "value") {
+      id
+    }
+  }
+`;
+
+const typeDefs = `
+type Stub {
+  id: String
+}
+
+type Query {
+  sampleQuery: Stub
+}
+`;
+
+const schema = makeExecutableSchema({ typeDefs });
+
+describe('SchemaLink', () => {
+  const data = { data: { hello: 'world' } };
+  const data2 = { data: { hello: 'everyone' } };
+  const mockError = { throws: new TypeError('mock me') };
+
+  let subscriber;
+
+  beforeEach(() => {
+    const next = jest.fn();
+    const error = jest.fn();
+    const complete = jest.fn();
+
+    subscriber = {
+      next,
+      error,
+      complete,
+    };
+  });
+
+  it('raises warning if called with concat', () => {
+    const link = new SchemaLink({ schema });
+    const _warn = console.warn;
+    console.warn = warning => expect(warning['message']).toBeDefined();
+    expect(link.concat((operation, forward) => forward(operation))).toEqual(
+      link,
+    );
+    console.warn = _warn;
+  });
+
+  it('throws if no arguments given', () => {
+    expect(() => new SchemaLink()).toThrow();
+  });
+
+  it('correctly receives the constructor arguments', () => {
```

Comment:
I renamed this test to match its behavior... I don't know how useful it is, but it doesn't hurt so I kept it around.